volk: 3.0.0 -> 3.1.2

doronbehar opened the pull request:
## Description of changes


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

doronbehar committed:
volk: 2.5.0 -> 3.0.0 (for aarch64-darwin only); move to pkgs/by-name

doronbehar committed:
volk: 3.0.0 -> 3.1.2; cleanup

Diff: https://github.com/gnuradio/volk/compare/v3.0.0...v3.1.2

doronbehar committed:
volk: nixfmt (rfc 166)

tko commented on pkgs/by-name/vo/volk/package.nix:
```diff
@@ -0,0 +1,46 @@
+    fetchSubmodules = true;
+  };
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/gnuradio/volk/commit/e46771a739658b5483c25ee1203587bf07468c4d.patch";
```
lost a comment here. is this patch still needed?

doronbehar replied:
> lost a comment here. is this patch still needed?

I forgot to comment, this patch is a different patch: https://github.com/gnuradio/volk/pull/766 .

doronbehar replied:
Now https://github.com/gnuradio/volk/pull/766 is mentioned in a comment.

tko commented on pkgs/by-name/vo/volk/package.nix:
```diff
@@ -0,0 +1,47 @@
+  src = fetchFromGitHub {
+    owner = "gnuradio";
+    repo = "volk";
+    rev = "v${finalAttrs.version}";
```
Should this be `"rev = v${version}";`

I realize it's probably copypasta from previous where versions were different, but is this normal in other cases?

doronbehar replied:
> Should this be `rev = "v${version}";`
> 
> I realize it's probably copypasta from previous where versions were different, but is this normal in other cases?

See:

https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

tko replied:
I'll take your word for it.

I can't see why this needs `finalAttrs` or to be a function rather than plain data (which I think are the options here)

doronbehar replied:
> I'll take your word for it.
> 
> I can't see why this needs `finalAttrs` or to be a function rather than plain data (which I think are the options here)

Now I see that the docs don't motivate people to switch to `finalAttrs`, which is a shame... In general the keypoints to do that are:

1. Easier overriding
2. Not having to use `rec` which is discouraged and is evaluation expensive (see [this](https://nix.dev/guides/best-practices#recursive-attribute-set-rec)).
3. Nicer expressions if you need to reference `finalAttrs.finalPackage`.

tko replied:
In general sure, all good points.

None apply here?

(I have zero experience with nix maintenance)

doronbehar replied:
> In general sure, all good points.
> 
> None apply here?

Why not? Only point number 3 doesn't apply.

tko commented:
FWIW I built `default.nix` on `aarch64-darwin` successfully earlier, and other than some cosmetics this PR seems fine.

`nixpkgs-review` is taking a long while...

doronbehar commented:
> FWIW I built `default.nix` on `aarch64-darwin` successfully earlier, and other than some cosmetics this PR seems fine.

That's not the same thing - no the expression is very different - it is not even located in the same place and it, and it doesn't build the same sources.

> `nixpkgs-review` is taking a long while...

Yea I should have recommended to you to wait a bit for ofborg to finish evaluation here so that you won't run it locally. See the ofborg (first) feature explained [here](https://github.com/Mic92/nixpkgs-review?tab=readme-ov-file#features). ofborg just finished evaluating attributes here, so respawning `nixpkgs-review` should at least start building things a bit faster. If it already builds attributes for you, there is no way to speed it up.

tko commented:
Result of `nixpkgs-review pr 330477` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>gnuradio3_8Packages.ais</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradio3_9</li>
    <li>gnuradioMinimal (gnuradio3_9Minimal)</li>
    <li>gnuradio3_9Packages.grnet</li>
    <li>gnuradio3_9Packages.osmosdr</li>
    <li>gnuradio3_9Packages.osmosdr.dev</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>sigutils</li>
    <li>suscan</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

doronbehar committed:
gnuradio3_9: try to fix build by disabling doxygen

Encounteres an error gnuradio3_8 also experiences when volkExternal ==
volk (3.1.2).

doronbehar committed:
gnuradio3_9{,Minimal}: remove

doronbehar commented:
Result of `nixpkgs-review pr 330477` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradio3_9</li>
    <li>gnuradio3_9Packages.grnet</li>
    <li>gnuradio3_9Packages.osmosdr</li>
    <li>gnuradio3_9Packages.osmosdr.dev</li>
    <li>inspectrum</li>
    <li>qradiolink</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradioMinimal (gnuradio3_9Minimal)</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>nixpkgs-manual</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>sigutils</li>
    <li>suscan</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

doronbehar commented:
Hmm very peculiar ... The `gnuradio3_8` builds failed for me on x86_64-linux. Also `gnuradio3_9` builds failed, but in a slightly different manner.

I pushed a few more commits for to these attributes. If @tie you'd be kind enough to run a `nixpkgs-review` again for these commits it'd be great, I can cc you when ofborg finished evaluation, so it won't take too much resources from you like before.

tie commented:
> If @tie you'd be kind enough to run a nixpkgs-review again

I assume this is a typo and you wanted to ping @tko 😅
But feel free to also ping me, I can run nixpkgs-review on `aarch64-linux`.

tko commented:
```
error:
       … while querying the derivation named 'gr-ais-2020-08-13'
[..]
       error: attribute 'volk' missing

       at /Users/tko/.cache/nixpkgs-review/pr-330477-6/nixpkgs/pkgs/development/gnuradio-modules/mkDerivation.nix:29:59:

           28|     # always be needed
           29|     buildInputs = (args.buildInputs or []) ++ [ unwrapped unwrapped.volk ];
             |                                                           ^
           30|   };
```

doronbehar committed:
gnuradio3_8Packages.ais: mark as broken on Darwin

doronbehar commented:
> ```
> error:
>        … while querying the derivation named 'gr-ais-2020-08-13'
> [..]
>        error: attribute 'volk' missing
> 
>        at /Users/tko/.cache/nixpkgs-review/pr-330477-6/nixpkgs/pkgs/development/gnuradio-modules/mkDerivation.nix:29:59:
> 
>            28|     # always be needed
>            29|     buildInputs = (args.buildInputs or []) ++ [ unwrapped unwrapped.volk ];
>              |                                                           ^
>            30|   };
> ```

Thanks. At least this error will be fixed now. (will recheck ofborg once more in a while).

doronbehar commented:
Evaluation finished successfully. Now running nixpkgs-review locally on an `x86_64-linux`.

doronbehar commented:
Result of `nixpkgs-review pr 330477` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>inspectrum</li>
    <li>nixpkgs-manual</li>
    <li>qradiolink</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>sigutils</li>
    <li>suscan</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

tko commented:
Result of `nixpkgs-review pr 330477` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnuradio3_8Packages.ais</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>sigutils</li>
    <li>suscan</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

doronbehar commented:
Thanks for the quick report! Could you share please the build logs of `gnuradio3_8Minimal`? It is available in `logs` directory of the `nixpkgs-review` shell..

tie commented:
Result of `nixpkgs-review pr 330477` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>gnss-sdr</li>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.ais</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.limesdr</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>gqrx</li>
    <li>gqrx-gr-audio</li>
    <li>gqrx-portaudio</li>
    <li>inspectrum</li>
    <li>qradiolink</li>
    <li>sdrpp</li>
    <li>sigdigger</li>
    <li>sigutils</li>
    <li>suscan</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

tie commented:
Had to run with `--build-args '--max-jobs 1'` because of OOM (with 32GB, 16 cores).

tko commented:
[gnuradio3_8Minimal.log](https://github.com/user-attachments/files/16403030/gnuradio3_8Minimal.log)

doronbehar committed:
volk_2: init at 2.5.0

Based on the expression deleted in a previous commit, use this
expression in an upcoming commit for gnuradio3_8{,Minimal}.

doronbehar committed:
gnuradio3_8: fix build by using volk_2

doronbehar commented:
Thanks @tko :). This build log I am afraid indicates that for GR 3.8 there is no choice but to go back to volk2, for all platforms. I've found for it an alternative build solution - using Python 3.11 (like GR 3.8 uses), as it still supports the deprecated `distutils`.

If in an hour or two (after ofborg evaluation finishes) you'd be able to run a `nixpkgs-review` again, that would be great :pray:.

tko commented:
Result of `nixpkgs-review pr 330477` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnuradio3_8Packages.ais</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>sigdigger</li>
    <li>suscan</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>gnuradio</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_8Minimal</li>
    <li>gnuradio3_8Packages.grnet</li>
    <li>gnuradio3_8Packages.osmosdr</li>
    <li>gnuradio3_8Packages.osmosdr.dev</li>
    <li>gnuradio3_8Packages.rds</li>
    <li>gnuradioMinimal</li>
    <li>gnuradioPackages.osmosdr</li>
    <li>gnuradioPackages.osmosdr.dev</li>
    <li>sdrpp</li>
    <li>sigutils</li>
    <li>suwidgets</li>
    <li>volk</li>
  </ul>
</details>

The build error looks to have been `suscan`, dependency of `sigdigger`
```
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY) (found version "1.2.11")
[..]
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ZLIB_LIBRARY
    linked by target "suscan" in directory /tmp/nix-build-suscan-unstable-2022-07-05.drv-0/source
```

doronbehar committed:
suscan: unstable-2022-07-05 -> 0.3.0

Nothing really changed, only upstream tagged the same commit as v0.3.0,
see:
https://github.com/BatchDrake/suscan/compare/37dad542b97aff24654f0bb80fb8e85af7cb84ab...v0.3.0

doronbehar commented:
Thanks again for the report @tko. Maybe an update to `suscan` will fix the issue. I checked that `sigdigger` compiles on `x86_64-linux` with the `suscan` update I just pushed, and I checked basic functionality of both packages' commands.

I will give a few days for the maintainers of both packages (@polygon and @oxapentane) to respond to this `suscan` update, and I will also let a few more days for ofborg to build this on `aarch64-darwin`. If you really want to speed things up, you can run a `nixpkgs-review` once more, but that's up to you :). Thanks again for contributing your local hardware's time :).

doronbehar commented:
@ofborg build sigdigger

tko commented:
I see same error from `suscan-0.3.0`
```
-- Could NOT find ZLIB (missing: ZLIB_LIBRARY) (found version "1.2.11")
[..]
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
ZLIB_LIBRARY
    linked by target "suscan" in directory /tmp/nix-build-suscan-0.3.0.drv-0/source
```

doronbehar commented:
OK I found out that the suscan update wasn't really an update - upstream only decided to tag the same revision that we used as v0.3.0 . Makes the expression nicer, but of course it won't solve the zlib issue.

I fixed the expression to use `rev = "v${version}";`. And I won't make an effort to fix that. It is barely related to this update, and I can't fix that without _my_ hands on a Darwin machine. I also expect Darwin users to experience this issue in the next channel update (if there are Darwin users that use this package).

I'll give a few days for the maintainers of `suscan` to respond, and I'll merge this in a few days.

doronbehar committed:
suscan: fix build by adding zlib

doronbehar commented:
OK suscan was broken apparently on all platforms - I missed that because my branch wasn't rebased upon latest `master`, and ofborg didn't miss that. Now it should be fixed. I'll wait for CI to be greener and I'll merge this.

SuperSandro2000 commented on pkgs/by-name/vo/volk2/package.nix:
```diff
@@ -0,0 +1,64 @@
+
+  nativeBuildInputs = [
+    cmake
+    python311
+    python311.pkgs.mako
```
Would be nice to comment why we can't use 3.12

doronbehar replied:
> Would be nice to comment why we can't use 3.12

Done.

SuperSandro2000 commented on pkgs/by-name/vo/volk2/package.nix:
```diff
@@ -0,0 +1,64 @@
+    ];
+
+  postInstall = lib.optionalString (!stdenv.isDarwin) ''
+    ${removeReferencesTo}/bin/remove-references-to -t ${stdenv.cc} $(readlink -f $out/lib/libvolk.so)
```
Please put removeReferencesTo into nativeBuildInputs

doronbehar replied:
> Please put removeReferencesTo into nativeBuildInputs

Good idea.

doronbehar commented on pkgs/by-name/vo/volk2/package.nix:
> That file should be in the volk_2 directory

Is there a convention that we prefer `volk_2` and not `volk2` as attribute names?

doronbehar replied:
> > That file should be in the volk_2 directory
> 
> Is there a convention that we prefer `volk_2` and not `volk2` as attribute names?

Did that.

doronbehar committed:
gnuradio{,3_8}: use removeReferencesTo from nativeBuildInputs
